tests/goth/test_run_yacat.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import os
+from pathlib import Path
+import re
+
+import pytest
+
+from goth.assertions import EventStream
+from goth.configuration import load_yaml
+from goth.runner.log import configure_logging
+from goth.runner import Runner
+from goth.runner.probe import RequestorProbe
+
+
+logger = logging.getLogger("goth.test.run_yacat")
+
+ALL_TASKS = {"0", "48", "None"}
+
+
+# Temporal assertions expressing properties of sequences of "events". In this case, each "event"
+# is just a line of output from `yacat.py`.
+
+
+async def assert_no_errors(output_lines: EventStream[str]):
+    """Assert that no output line contains the substring `ERROR`."""
+    async for line in output_lines:
+        if "ERROR" in line:
+            raise AssertionError("Command reported ERROR")
+
+
+async def assert_all_tasks_processed(status: str, output_lines: EventStream[str]):
+    """Assert that for every task in `ALL_TASKS` a line with `Task {status}` will appear."""
+    remaining_tasks = ALL_TASKS.copy()
+
+    async for line in output_lines:
+        m = re.search(rf".*Task {status} .* task data: ([a-zA-Z0-9]+)$", line)
+        if m:
+            task_data = m.group(1)
+            logger.debug("assert_all_tasks_processed: Task %s: %s", status, task_data)
+            remaining_tasks.discard(task_data)
+        if not remaining_tasks:
+            return
+
+    raise AssertionError(f"Tasks not {status}: {remaining_tasks}")
```

Comment:
The same function is defined in `test_run_blender.py`. We should make `ALL_TASKS` a parameter and move the function to a separate module so it could be imported in both `test_run_blender.py` and `test_run_yacat.py`.

The same goes for `assert_no_errors` and `assert_all_invoices_accepted`.

Reply:
We can also do it in another PR, the current one is already over 2 weeks old.

Reply:
This sounds more pythonically complicated than just copy-n-adjust I can do ;)
I will add an issue for this in the backlog, and merge this PR as is.